packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -192,37 +198,72 @@ export default function __experimentalUseColors(
 	);
 
 	const detectedBackgroundColorRef = useRef();
-	const BackgroundColorDetector = useMemo(
-		() =>
-			contrastCheckers &&
-			( Array.isArray( contrastCheckers ) ?
-				contrastCheckers.some(
-					( { backgroundColor } ) => backgroundColor === true
-				) :
-				contrastCheckers.backgroundColor === true ) &&
-			withFallbackStyles( ( node, { querySelector } ) => {
-				if ( querySelector ) {
-					node = node.parentNode.querySelector( querySelector );
-				}
-				let backgroundColor = getComputedStyle( node ).backgroundColor;
-				while ( backgroundColor === 'rgba(0, 0, 0, 0)' && node.parentNode ) {
-					node = node.parentNode;
-					backgroundColor = getComputedStyle( node ).backgroundColor;
+	const detectedColorRef = useRef();
+	const ColorDetector = useMemo( () => {
+		let needsBackgroundColor = false;
+		let needsColor = false;
+		for ( const { backgroundColor, textColor } of castArray( contrastCheckers ) ) {
+			if ( ! needsBackgroundColor ) {
+				needsBackgroundColor = backgroundColor === true;
+			}
+			if ( ! needsColor ) {
+				needsColor = textColor === true;
+			}
+			if ( needsBackgroundColor && needsColor ) {
+				break;
+			}
+		}
+		return (
+			( needsBackgroundColor || needsColor ) &&
+			withFallbackStyles(
```

Comment:
We need to re-execute the color detection when a color challenges. On the paragraph block, we want to automatically detect background and text color `contrastCheckers: { backgroundColor: true, textColor: true, fontSize: fontSize.size },`, we can also change both colors. The contrast warnings don't update when a color challenges but if we go to the code editor and switch back the visual editor to force the component to remount the warnings appear as expected.

Reply:
This seems to be working after your last rebase?

Reply:
No, I think the warnings still don't appear when expected:
![Nov-26-2019 09-04-09](https://user-images.githubusercontent.com/11271197/69614897-e469d280-102b-11ea-8965-5ec74df013f2.gif)

We can see the warning appears when it should not, and disappears when it should appear. If I type something and the warning should appear, it appears after I type. Not sure why typing interferes with this, but we should make sure the color computation does not happen on each character typed.

Reply:
This was happening because setting a ref does not re-render the component, so the contrast checker used a stale value until the component had any sort of update.

Fixed: https://github.com/WordPress/gutenberg/pull/18547/commits/45c81d4a629f005ab4e9557d415e189ef4d844ab.